Tests.Unit.Core.ApplicationServices/ApplicationServices/Deltas/TrackingServiceTests.cs

```diff
@@ -0,0 +1,205 @@
+{
+    public class TrackingServiceTests : WithAutoFixture
+    {
+        private readonly TrackingService _sut;
+        private readonly Mock<IGenericRepository<LifeCycleTrackingEvent>> _repositoryMock;
+        private readonly Mock<IAuthorizationContext> _authorizationContextMock;
+        private readonly Mock<IOrganizationalUserContext> _organizationUserContextMock;
+
+        public TrackingServiceTests()
+        {
+            _repositoryMock = new Mock<IGenericRepository<LifeCycleTrackingEvent>>();
+            _authorizationContextMock = new Mock<IAuthorizationContext>();
+            _organizationUserContextMock = new Mock<IOrganizationalUserContext>();
+            _sut = new TrackingService(_repositoryMock.Object, _authorizationContextMock.Object, _organizationUserContextMock.Object);
+        }
+
+        protected override void OnFixtureCreated(Fixture fixture)
+        {
+            var outer = new Fixture();
+            fixture.Inject<AccessModifier?>(AccessModifier.Local);
+            fixture.Register(() => new Organization
+            {
+                Id = outer.Create<int>(),
+                Uuid = outer.Create<Guid>()
+            });
+        }
+
+        [Fact]
+        public void Can_Query_With_No_Refinements_As_Full_CrossLevelAccess()
+        {
+            //Arrange
+            var all = Many<LifeCycleTrackingEvent>().AsQueryable();
+            _repositoryMock.Setup(x => x.AsQueryable()).Returns(all);
+            ExpectCrossLevelOrganizationReadAccess(CrossOrganizationDataReadAccessLevel.All);
+
+            //Act
+            var result = _sut.QueryLifeCycleEvents();
+
+            //Assert
+            Assert.Same(all, result);
+        }
+
+        [Fact]
+        public void Can_Query_With_No_Refinements_As_Public_CrossLevelAccess()
+        {
+            //Arrange
+            var uniqueIds = CreateUniqueIds(12).ToList();
+            var orgIds = uniqueIds.Take(2).ToList();
+            var idsToSetOnFullSet = new Stack<int>(uniqueIds.Skip(2));
+            var all = Many<LifeCycleTrackingEvent>(10).AsQueryable();
+
+            all.ToList().ForEach(x => x.OptionalOrganizationReference.Id = idsToSetOnFullSet.Pop());
+            all.Take(2).ToList().ForEach(x => x.OptionalOrganizationReference.Id = orgIds.First());
+            all.Skip(2).Take(2).ToList().ForEach(x => x.OptionalOrganizationReference.Id = orgIds.Last());
+            all.Skip(6).Take(2).ToList().ForEach(x => x.OptionalAccessModifier = AccessModifier.Public);
+
+            _organizationUserContextMock.Setup(x => x.OrganizationIds).Returns(orgIds);
+            _repositoryMock.Setup(x => x.AsQueryable()).Returns(all);
+            ExpectCrossLevelOrganizationReadAccess(CrossOrganizationDataReadAccessLevel.Public);
+
+            //Act
+            var result = _sut.QueryLifeCycleEvents();
+
+            //Assert - first 4 by org id relation, then two are filtered off due to visibility, then two are included because they are shared
```

Comment:
Du mener vel: 
first 4 by org id relation, then _**six**_ are filtered off due to visibility, then two are included because they are shared

Reply:
Nej
![image](https://user-images.githubusercontent.com/13116889/134680013-9d194b16-a1c9-4e1e-8ab8-749054fbbcf9.png)
Udgangspunktet for alle "skip" er position 0 :-)